### ISSUE TITLE
Training and testing the model

### ISSUE THREAD

**Issue by privefl** [gabraham/SparSNP#1]
I would like to include SparSNP in a comparison of statistical methods for computing polygenic risk scores.
In the documentation, you show how to perform a cross-validation, but I didn't find how to only train the model with some part of the data, and predict values for the rest of the data. In other words, how to perform the core operations within the cross-validation.
Is it possible? How to do it?

Thanks for your help.

**Comment by gabraham** [comment]
Hi @privefl,

You can split the data using plink, then train a model on the training dataset using the script https://github.com/gabraham/SparSNP/blob/master/runonce.sh. Then you can predict on new data using `plink --score `, which will take care of the issue of the reference allele being consistent between training and test.

```bash
export PATH=$PATH:~/Code/sparsnp

~/Code/sparsnp/runonce.sh training_data linear 0.1

awk -F',|\t| ' 'NR == FNR {a[$1]=$2; next} FNR in a {print $2, $5, a[FNR]}' \
   beta.csv.01.00 training_data.bim > score.txt

plink --bfile test_data --score score.txt
```

This won't take into account the intercept term, but for most uses it doesn't matter anyway.

Also, you don't get a whole range of L1 penalties this way, just one.

Hope this helps.

**Comment by privefl** [comment]
Hi again, 

I can now work fully on this. 
So, reading the documentation, I tried using `crossval.sh {tmpfile} sqrhinge` where {tmpfile} is replaced by the root of bed/bim/fam files.
In directory `crossval1` for example, I get files `y.00` to `y.09` with values which are all `1.0000000000` (my training set is of 6000 samples on the celiac disease). Is there a problem?
***
I would like to run only one run of cross-validation which get me the best model out of it (or 10 models, that I could average) so that I can get the predictions on other bed/bim/fam files (with same order, etc).

Could you help me please?

**Comment by privefl** [comment]
Thanks for your help.

I would like to run one cross-validation to get the best model out of it.
Then I want to get predictions for this particular model for independent samples (I will take care of the splitting of PLINK files).

How can I do this?

**Comment by gabraham** [comment]
Did you figure out the why the y.0X files were all 1.0? This isn't right,
there should be -1 there as well. Is your data in the right format, i.e.
1/2 for controls/cases in the fam file?

To get predictions out just for this one model, use the awk + PLINK I
mentioned earlier to match the positions of the SNPs to their rsids in your
data:

awk -F',|\t| ' 'NR == FNR {a[$1]=$2; next} FNR in a {print $2, $5, a[FNR]}' \
   beta.csv.01.00 training_data.bim > score.txt

plink --bfile test_data --score score.txt

**Comment by privefl** [comment]
The all 1s were an error of my part.

Is it okay with you if I consider the model of SparSNP to be give similar predictions than a logistic regression with elastic-net regularization?
Because, unfortunately, I won't be able to test every possible methods.

**Comment by gabraham** [comment]
In my experience, the prediction performance of the squared hinge loss with
L1 penalty is very similar to that of logistic regression with L1 penalty.
The main difference is that you can't interpret the predictions as log
odds, if that matters.
